### DISCUSSION TITLE
Speed of rpi 5 compiling a timelapse?

### DISCUSSION THREAD

**Discussion by ikeysolomon** [aaronwmorris/indi-allsky#1281] [Q&A]
Hi I currently have a rpi 3b+ that I have set up. However compiling a time-lapse on board is a bridge too far. Looking at a rpi 5 with 8gb. Just wondering what sort of times people get with using a rpi5 with 8gb to compile using the pi itself? Thank you.

**Comment by aaronwmorris** [comment]
The video compile time is probably going to be 2-3 times faster with a Pi4 or Pi5.  It benefits both from having more memory and faster CPUs.  Anything more than 4GB of memory is overkill in most situations.

You also need adequate cooling.  ffmpeg uses 100% of all cores and the system will quickly heat up.  A heat sink and fan will be needed to prevent throttling.

**Comment by ikeysolomon** [comment]
Given that I can't complete it on the 3b+, what time are we looking at here? 3x0 still = 0 haha

**Reply by aaronwmorris** [reply]
Hah.  The primary problem most people run into with the Raspi 3 is memory.  The CPUs are slower, but memory is key for the timelapse.  The amount of memory required by ffmpeg to generate the timelapse is directly proportional to the resolution of the images.  If you are using the ASI120, the 3b usually has enough, but you need to make sure you are running a non-desktop GUI enabled system and you may still need 512MB to 1GB of swap still to prevent running out of memory.

You can disable the desktop mode by running the following and reboot.
```
sudo systemctl set-default multi-user.target
```

**Comment by ikeysolomon** [comment]
Yeah I've done this. Also put a fan and a heat sink in the thing but still to no avail. You're right the ram is killing me here. I left the thing go the other day and it took over 2 hours and got no where. I'm pretty sure that when I installed the software it wanted to allocate a huge swap file. But yeah 2 hours is intolerable. I'm expecting alot of it. I will compile myself through loading on and off a usb. What should a pi4 with 4gb say do a nights worth of jpgs in?

**Reply by aaronwmorris** [reply]
2 hours does seem excessive given your situation.  My cameras are a RockPi3 and a LePotato which are not much faster than a Pi3, but they both have 2GB of RAM.  My cameras are a little higher resolution than yours (1920x1080).

This morning, one of my cameras generated a video from 1981 images in 17 minutes.  The other 1501 images in 11 minutes.
